temba/api/support.py

```diff
@@ -41,6 +42,33 @@ def authenticate_credentials(self, key):
         raise exceptions.AuthenticationFailed("User inactive or deleted")
 
 
+class APIBasicAuthentication(BasicAuthentication):
+    """
+    Basic authentication.
+
+    Clients should authenticate using HTTP Basic Authentication.
+
+    Credentials: username:api_token
+    """
+
+    def authenticate_credentials(self, userid, password):
+        try:
+            token = APIToken.objects.get(is_active=True, key=password)
+        except APIToken.DoesNotExist:
+            raise exceptions.AuthenticationFailed("Invalid token")
+
+        if token.user.username != userid:
+            raise exceptions.AuthenticationFailed("Invalid username")
+
+        if token.user.is_active:
+            # set the org on this user
+            token.user.set_org(token.org)
+
+            return token.user, token
+
+        raise exceptions.AuthenticationFailed("User inactive or deleted")
```

Comment:
These should all return "Invalid Token" otherwise you are leaking information. (like that the username doesn't exist in the system for example)

Reply:
(can change the TokenAuthentication above to do the same thing)